core/shared/src/main/scala/fs2/interop/flow/package.scala

```diff
@@ -199,6 +198,51 @@ package object flow {
   ): Stream[F, Nothing] =
     StreamSubscription.subscribe(stream, subscriber)
 
+  /** Creates a [[Processor]] from a [[Pipe]].
+    *
+    * You are required to manually subscribe this [[Processor]] to an upstream [[Publisher]], and have at least one downstream [[Subscriber]] subscribe to the [[Consumer]].
+    *
+    * Closing the [[Resource]] means not accepting new subscriptions,
+    * but waiting for all active ones to finish consuming.
+    * Canceling the [[Resource.use]] means gracefully shutting down all active subscriptions.
+    * Thus, no more elements will be published.
+    *
+    * @see [[unsafePipeToProcessor]] for an unsafe version that returns a plain [[Processor]].
+    *
+    * @param pipe The [[Pipe]] which represents the [[Processor]] logic.
+    * @param chunkSize setup the number of elements asked each time from the upstream [[Publisher]].
+    *                  A high number may be useful if the publisher is triggering from IO,
+    *                  like requesting elements from a database.
+    *                  A high number will also lead to more elements in memory.
+    */
+  def pipeToProcessor[F[_], I, O](
```

Comment:
We can add these as syntax on `Pipe` too :)

https://github.com/typelevel/fs2/blob/0377f21532f3486c54da62d38430d9657d052960/core/shared/src/main/scala/fs2/Stream.scala#L5494